### DISCUSSION TITLE
Retrieving HMTL content as is

### DISCUSSION THREAD

**Discussion by ramyabitorder** [SalesforceCommerceCloud/sfcc-ci#546] [Team Posts]
Hello All,

We have a custom attribute in our content asset which contains HTML contents, example below. When we try to extract the content, all the $url with the html content are getting replace with the actual url . Code snippet to extract added below. Need help to extract the values as is without replaced urls. 

Thanks a ton in advance for all the support. Let me know if further details are required .

html content :

href="$url('Product-Show','pid','CHDHX-701-xxxx')$"  in html , retrieved as

href=https://instance.demandware.net/on/demandware.store/Sites-Site/default/Product-Show?pid=CHDHC-601-xxxx”

Code snippet

   assetObj.body = eachContentAsset.custom.body;  --->returns  href=https://instance.demandware.net/on/demandware.store/Sites-Site/default/Product-Show?pid=CHDHC-601-xxxx” within the html contents instead of href="$url('Product-Show','pid','CHDHX-701-xxxx')$"  which is as is.

**Comment by pravngaur** [comment]
Hi,

This is happening because the custom attribute that you have created is
probably HTML type. Which is why platform is converting the html tags and
all while returning the API response.
If you want raw text as response then try using the attribute type of text
only, not html.

Give it a try.

**Comment by ramyabitorder** [comment]
Hello Praveen,

Thank you so much for the timely response. 

We have to introduce another plain text field, manually copy the value from this HTML BODY text field into it and then extract the value from the new text field where $url are not replaced with actually . We had to do this in order to not break the existing functionality. 
We anticipate another issue now and need your inputs on it. We manipulate the contents of the new plain text field and have to write it back to the HTML BODY field. During this step also i believe the $url s in the plain text will get replaced to full url. 
Can you please suggest if there is something we are missing and there is another api to do this.

**Comment by pravngaur** [comment]
Hi, i am little confused, why do we need two redundant attributes?

Please explain your use case, what are you trying to achieve? And also how do you use this attribute?

**Comment by ramyabitorder** [comment]
Hello Praveen,

Sorry I wasn’t clear enough.

We have this BODY field in content asset. We have to send this body text to a server for updation and write the updated value back to the same field on server response. This is all happening correctly. However during the content extraction process, the $URL fields in the content are getting replaced with actual urls. 

As per your suggestion. we couldn’t change the datatype of the BODY field. So we introduced a new field, copied the value from BODY field manually and used its value to send to server. 

On return from server, we need to write the updated value from server  back to both HTML and  Plain text field. Again when we copy to the BODY field the $urls are getting replaced. 

Nee your assistance to point to the right api where $url don’t get transformed to actual urls  while populating the HTML field.

**Comment by riconowak** [comment]
Hi

If you are able to use OCAPI, you can try the MarkupText API. Check out https://documentation.b2c.commercecloud.salesforce.com/DOC1/topic/com.demandware.dochelp/OCAPI/19.5/data/Documents/MarkupText.html?cp=0_12_6_100

Am 13.05.19 um 07:11 schrieb ramyabitorder:

Hello Praveen,

Sorry I wasn’t clear enough.

We have this BODY field in content asset. We have to send this body text to a server for updation and write the updated value back to the same field on server response. This is all happening correctly. However during the content extraction process, the $URL fields in the content are getting replaced with actual urls.

As per your suggestion. we couldn’t change the datatype of the BODY field. So we introduced a new field, copied the value from BODY field manually and used its value to send to server.

On return from server, we need to write the updated value from server back to both HTML and Plain text field. Again when we copy to the BODY field the $urls are getting replaced.

Nee your assistance to point to the right api where $url don’t get transformed to actual urls while populating the HTML field.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/orgs/SalesforceCommerceCloud/teams/community/discussions/35/comments/4>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AD2XAJEMFPE4Y7M5EDLCKS3PVD2A7ANCNFSM4HMIX5PA>.

--

Mit freundlichem Gruß / Regards

Rico Nowak
Developer


holosystems GmbH & Co. KG
Hauptstrasse 16
99955 Herbsleben
Tel.: 036041 50914
Fax: 036041 50919
Mail: rico@holosystems.de<mailto:rico@holosystems.de>
URL: http://www.holosystems.de


_______________________________________________________________________________________________________
Bitte denken Sie über Ihre Verantwortung gegenüber der Umwelt nach, bevor Sie diese E-Mail ausdrucken!
Please consider the environment before printing this e-mail!

**Comment by ramyabitorder** [comment]
> Hi
> 
> If you are able to use OCAPI, you can try the MarkupText API. Check out https://documentation.b2c.commercecloud.salesforce.com/DOC1/topic/com.demandware.dochelp/OCAPI/19.5/data/Documents/MarkupText.html?cp=0_12_6_100
> 
> Am 13.05.19 um 07:11 schrieb ramyabitorder:
> 
> Hello Praveen,
> 
> Sorry I wasn’t clear enough.
> 
> We have this BODY field in content asset. We have to send this body text to a server for updation and write the updated value back to the same field on server response. This is all happening correctly. However during the content extraction process, the $URL fields in the content are getting replaced with actual urls.
> 
> As per your suggestion. we couldn’t change the datatype of the BODY field. So we introduced a new field, copied the value from BODY field manually and used its value to send to server.
> 
> On return from server, we need to write the updated value from server back to both HTML and Plain text field. Again when we copy to the BODY field the $urls are getting replaced.
> 
> Nee your assistance to point to the right api where $url don’t get transformed to actual urls while populating the HTML field.
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub<https://github.com/orgs/SalesforceCommerceCloud/teams/community/discussions/35/comments/4>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AD2XAJEMFPE4Y7M5EDLCKS3PVD2A7ANCNFSM4HMIX5PA>.
> …

Hello Rico,

Thank you so much. This helped. We used the getSource API to retrieve content as is from the documentation you shared.
https://documentation.b2c.commercecloud.salesforce.com/DOC1/index.jsp?topic=%2Fcom.demandware.dochelp%2FDWAPI%2Fscriptapi%2Fhtml%2Fapi%2Fclass_dw_content_MarkupText.html&cp=0_16_2_4_8
